dali/dfuXRefLib/dfuxreflib.cpp

```diff
@@ -335,8 +335,9 @@ static unsigned short getDafsPort(const SocketEndpoint &ep,unsigned &numfails,Cr
     }
     else if (numfails>5)
         return 0;
-    if (testDaliServixPresent(ep)) 
-        return ep.port?ep.port:getDaliServixPort();
+    unsigned short nPort = ep.port;
+    if (testDaliServixPresent(ep, &nPort))
+        return nPort?nPort:getDaliServixPort();
```

Comment:
the combination of a non-cost SocketEndpoint purely so other method can reuse, that changes port _and_ port is altered via passed in ptr isn't very nice.
You can change checkLocalDaFileSvr() to call a cleaner method e.g. getActiveDaliServixPort. I made a comment earlier in checkLocalDaFileSvr() re. that.